nemo/collections/asr/parts/submodules/ctc_decoding.py

```diff
@@ -858,6 +902,92 @@ def _get_word_offsets_subwords_sentencepiece(
 
         return word_offsets
 
+    @staticmethod
+    def _get_segment_offsets(
+        offsets: Dict[str, Union[str, float]],
+        segment_delimiter_tokens: List[str],
+        supported_punctuation: Set = None,
+        segment_gap_threshold: int = None,
+    ) -> Dict[str, Union[str, float]]:
+        """
+        Utility method which constructs segment time stamps out of word time stamps.
+
+        Args:
+            offsets: A list of dictionaries, each containing "word", "start_offset" and "end_offset".
+            segments_delimiter_tokens: List containing tokens representing the seperator(s) between segments.
+            supported_punctuation: Set containing punctuation marks in the vocabulary.
+            segment_gap_threshold: Number of frames between 2 consecutive words necessary to form segments out of plain text.
+
+        Returns:
+            A list of dictionaries containing the segment offsets. Each item contains "segment", "start_offset" and
+            "end_offset".
+        """
+        if (
+            supported_punctuation
+            and not set(segment_delimiter_tokens).intersection(supported_punctuation)
+            and not segment_gap_threshold
+        ):
+            logging.warning(
```

Comment:
Why segment_gap_threshold condition is checked here? 
Log this only once using logging MODE=ONCE, otherwise we get this logging for each utterance.